src/HttpClientSample.Test/HttpClientSampleTests.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Net.Http;
+using Autofac;
+using FluentAssertions;
+using HttpClientSample.Core;
+using Xunit;
+
+namespace HttpClientSample.Test
+{
+    public class HttpClientSampleTests
+    {
+        [Fact]
+        public void ContainerShouldAlwaysReturnTheSameHttpClient()
```

Comment:
2

Reply:
✅ ArrayUtils provide many thread-safe operations on arrays increasing the developer productivity. This class tries to handle `null` input gracefully. An exception will not be thrown for a `null` array input. However, an Object array that contains a `null` element may throw an exception.

**No change suggested, looks good!**

Reply:
✅ ArrayUtils provide many thread-safe operations on arrays increasing the developer productivity. This class tries to handle `null` input gracefully. An exception will not be thrown for a `null` array input. However, an Object array that contains a `null` element may throw an exception.

**No change suggested, looks good!**
          
## The solution in this case

```Java
public class Name {
}
```

Reply:
❗️ The declaration and name of this `{Set/List/Map}` suggest an immutable collection, but Guava, {Sets.newHashSet()/ Lists.newArrayList()/ Maps.newHashMap()} method creates a mutable, initially empty instance of this collection. Therefore, this snippet of code is prone to share mutability and is misleading to its clients. It is recommended to rely on Guava, Immutable{Set/List/Map}.of() as in the example.  

[**More details**](https://confluence.devfactory.com/x/6hJrEw)
          
## The solution in this case

```Java

```

Reply:
⚠️  An improvement could be to simplify this class (and similar classes) by using features provided by [Project Lombok](https://projectlombok.org/). By adding the proper annotations for constructors, setters, getters, etc., Lombok will automatically generate the corresponding code.  

[**More details**](https://confluence.devfactory.com/x/-xCUEw)
          
## The solution in this case

```Java

```

Reply:
❗️ `{Optional if using @Data}`: Since `@Data` is a shortcut for adding both `@Getter` and `@Setter` on class-level, the following applies:
Using `@Getter` (and/or `@Setter`), Lombok will generate public accessors for all the non-static fields (even package-protected, protected and public ones), so it is recommended to tighten encapsulation to private and use the accessors within possible child classes and clients, instead of actual fields. Doing so, there is only one way of accessing the data, which leads to better code flow and consistency.  

[**More details**](https://confluence.devfactory.com/x/iQhrEw)
          
## The solution in this case

```Java

```